PowerPointLabs/PowerPointLabs/TimerLab/TimerLabPaneWPF.xaml.cs

```diff
@@ -655,13 +786,77 @@ private void HeightTextBox_LostFocus(object sender, RoutedEventArgs e)
         #endregion
 
         #region Timer Helper
+        private void ChangeTimerBodyColor(int timerBodyColor)
+        {
+            timerBody.Fill.ForeColor.RGB = timerBodyColor;
+            timerBody.Line.ForeColor.RGB = timerBodyColor;
+        }
+
+        private void CreateProgressBarShape(float timerWidth, float timerHeight, int sliderColor)
+        {
+            // Creation of progress bar
+            Shape visibleProgressBar = this.GetCurrentSlide().Shapes.AddShape(Microsoft.Office.Core.MsoAutoShapeType.msoShapeRectangle,
+                                                                       0, 0, timerWidth, timerHeight);
+            visibleProgressBar.Fill.ForeColor.RGB = sliderColor;
+            visibleProgressBar.Line.ForeColor.RGB = sliderColor;
+            float rightOfProgressBar = visibleProgressBar.Left + visibleProgressBar.Width;
+            float topOfProgressBar = visibleProgressBar.Top;
+            // Position invisible progress bar to the right of the visible one
+            Shape invisibleProgressBar = this.GetCurrentSlide().Shapes.AddShape(Microsoft.Office.Core.MsoAutoShapeType.msoShapeRectangle,
+                                                                       rightOfProgressBar, topOfProgressBar, timerWidth, timerHeight);
+            invisibleProgressBar.Fill.Transparency = TimerLabConstants.TransparencyTransparent;
+            invisibleProgressBar.Line.Transparency = TimerLabConstants.TransparencyTransparent;
+            // Grouping the shapes together
+            int[] indicesOfProgressBars = new int[2];
+            Shapes shapesInSlide = this.GetCurrentSlide().Shapes;
+            for (int i = 1; i <= shapesInSlide.Count; i++)
+            {
+                if (shapesInSlide[i].Equals(visibleProgressBar))
+                {
+                    indicesOfProgressBars[0] = i;
+                    continue;
+                }
+
+                if (shapesInSlide[i].Equals(invisibleProgressBar))
+                {
+                    indicesOfProgressBars[1] = i;
+                    continue;
+                }
+            }
+            progressBar = (this.GetCurrentSlide().Shapes.Range(indicesOfProgressBars)).Group();
```

Comment:
Instead of looping through all the shapes in the slide, you might want to try the following to group the two shapes:

```c#
string[] barNames = new string[] { visibleProgressBar.Name, invisibleProgressBar.Name };
progressBar = this.GetCurrentSlide().Shapes.Range(barNames ).Group();
```

I believe passing the array of names into the `.Range()` method also works. This way we can shorten the code and also make it more efficient :)

Reply:
Thanks @blewjy it works! Previously I tried passing it as a `Shape[]` which did not work out :(